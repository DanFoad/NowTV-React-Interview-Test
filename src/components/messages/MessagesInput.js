import React from 'react'
import PropTypes from 'prop-types'

import './Messages.css'

class MessagesInput extends React.Component {
    constructor(props) {
        super(props)

        this.state = {
            text: ""
        }

        this.isEnterPress = this.isEnterPress.bind(this)
        this.updateText = this.updateText.bind(this)
        this.submitText = this.submitText.bind(this)
    }

    getActionIcons() {
        return (
            <div className="messages-input__actions">
                <img src="/emoticon.png" alt="" />
                <img src="/attachment.png" alt="" />
            </div>
        )
    }

    isEnterPress(event) {
        if (event.keyCode === 13 && event.shiftKey === false) {
            event.preventDefault()
            this.submitText(this.textarea.value)
        }
    }

    updateText(event) {
        this.setState({ text: event.target.value })
    }

    submitText(event) {
        this.props.sendCallback(this.state.text)
        this.setState({ text: "" })
    }

    render() {
        return (
            <div className="messages-input">
                <textarea
                    ref={ el => this.textarea = el }
                    className="messages-input__textarea"
                    placeholder="Write a reply..."
                    value={ this.state.text }
                    onChange={ this.updateText }
                    onKeyUp={ this.isEnterPress }
                />
                { this.getActionIcons() }
                <button onClick={ this.submitText } className="messages-input__send">Send</button>
            </div>
        )
    }
}

MessagesInput.propTypes = {
    sendCallback: PropTypes.func,
}

export default MessagesInput