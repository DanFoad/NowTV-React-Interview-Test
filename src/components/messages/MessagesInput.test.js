import React from "react";
import renderer from "react-test-renderer/";

import MessagesInput from "./MessagesInput";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MessagesInput />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});