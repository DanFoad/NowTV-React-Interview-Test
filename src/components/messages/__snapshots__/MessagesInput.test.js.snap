exports[`test renders correctly 1`] = `
<div
  className="messages-input">
  <textarea
    className="messages-input__textarea"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Write a reply..."
    value="" />
  <div
    className="messages-input__actions">
    <img
      alt=""
      src="/emoticon.png" />
    <img
      alt=""
      src="/attachment.png" />
  </div>
  <button
    className="messages-input__send"
    onClick={[Function]}>
    Send
  </button>
</div>
`;
